Apps/Src/SendTritium.c

```diff
@@ -0,0 +1,442 @@
+    else if(cruiseEnableCounter == 0){cruiseEnableButton = false;}
+
+    if(cruiseSetCounter == DEBOUNCE_PERIOD){cruiseSet = true;}
+    else if(cruiseSetCounter == 0){cruiseSet = false;}
+
+    // TOGGLE
+    if(ChargeEnable && regenButton != regenPrevious && regenPrevious){regenToggle = !regenToggle;}
+    regenPrevious = regenButton;
+    
+    if(cruiseEnableButton != cruiseEnablePrevious && cruiseEnablePrevious){cruiseEnable = !cruiseEnable;}
+    cruiseEnablePrevious = cruiseEnableButton;
+
+    #endif
+}
+
+/**
+ * @brief Converts integer percentage to float percentage
+ * @param percent integer percentage from 0-100
+*/
+extern const float pedalToPercent[];
+static float percentToFloat(uint8_t percent){
+    if(percent > 100){
+        return 1.0f;
+    }
+    return pedalToPercent[percent];
+}
+
+/**
+ * @brief Follows the FSM to update the velocity of the car
+*/
+void Task_SendTritium(void *p_arg){
+    OS_ERR err;
+    Minion_Error_t minion_err;
+
+    // Initialize current state to NORMAL_DRIVE
+    state = FSM[NORMAL_DRIVE];
+
+    #ifndef __TEST_SENDTRITIUM
+    CANPayload_t velocityPayload;
+    velocityPayload.bytes = 32;
+    velocityPayload.data.d = velocitySetpoint;
+
+    CANPayload_t currentPayload;
+    currentPayload.bytes = 32;
+    currentPayload.data.d = currentSetpoint;
+    #endif
+
+    while(1){
+        state.stateHandler();    // do what the current state does
+        readInputs();   // read inputs from the system
+        state.stateDecider();    // decide what the next state is
+
+        // Sets brakelight
+        Minion_Write_Output(BRAKELIGHT, brakelightState, &minion_err);
+
+        // Drive
+        #ifndef __TEST_SENDTRITIUM
+        if(MOTOR_MSG_COUNTER_THRESHOLD == motorMsgCounter){
+            motorMsgCounter = 0;
+
+            CANbus_Send(VELOCITY, velocityPayload, CAN_NON_BLOCKING);
+            CANbus_Send(CURRENT_VEC, currentPayload, CAN_NON_BLOCKING);
+
+        }else{
+            motorMsgCounter++;
+        }
+        #endif
+
+        // Dump Info for Testing
+        #ifdef __TEST_SENDTRITIUM
+        dumpInfo();
+        #endif
+
+        // Delay of few milliseconds (100)
+        OSTimeDlyHMSM(0, 0, 0, MOTOR_MSG_PERIOD, OS_OPT_TIME_HMSM_STRICT, &err);
+        if (err != OS_ERR_NONE){
+            assertOSError(OS_UPDATE_VEL_LOC, err);
+        }
+    }
+}
+
+// State Handlers & Deciders
+
+/**
+ * Normal Drive State. Accelerator is mapped directly to current setpoint.
+*/
+void NormalDriveHandler(){
+    if(forwardGear){
+        velocitySetpoint = MAX_VELOCITY;
+        currentSetpoint = percentToFloat(accelPedalPercent);
+    }else if(reverseGear){
+        velocitySetpoint = -MAX_VELOCITY;
+        currentSetpoint = percentToFloat(accelPedalPercent);
+    }else if(neutralGear){
+        velocitySetpoint = MAX_VELOCITY;
+        currentSetpoint = 0;
+    }
+}
+
+void NormalDriveDecider(){
+    if(brakePedalPercent >= NORMAL_BRAKE_THRESHOLD){
+        state = BRAKE_STATE;
```

Comment:
I'm guessing it's supposed to do this
```suggestion
        state = FSM[BRAKE_STATE];
```